### PR TITLE
feat: update mailgun response and store retry attempt in separate table

### PR DESCRIPTION
**Pull request by Harish-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed preliminary testing using the [test suite](../../apps/api/OsmoX.postman_collection.json) to ensure that any existing features are not impacted and any new features are working as expected.
- [ ] I have updated the required [api docs](../../apps/api/docs/) as applicable.
- [ ] I have added/updated test cases to the [test suite](../../apps/api/OsmoX.postman_collection.json) as applicable

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login endpoint`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [x] Screenshots have been added (optional)
- [ ] Query request and response examples have been added (as applicable, in case added or updated)
- [ ] Documentation changes have been listed (as applicable)
- [ ] Test suite output is added (as applicable)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

- This PR update the result stored for mailgun in the database - include all information earlier some info are missed
- Also update the Logic for storing the retry attempts in separate table

**Related changes:**

- Create a new entity `RetryNotification`
- Updated the provider to include retry Repository 
- Updated the notification service for adding the retry attempt to the `notify_notification_retrires` table
- add migration for the new table

**Screenshots:**

image

**Query request and response:**

NA

**Documentation changes:**

NA

**Test suite output:**

NA

**Pending actions:**

NA

**Additional notes:**

NA

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `RetryNotification` entity to manage retry attempts for notifications.
	- Enhanced notification handling across various consumers to incorporate retry logic, improving robustness and error management.
	- Added logic to save retry attempts in the notification processing workflow, allowing for better tracking and logging of delivery issues.
	- Added a new method in the `NotificationsService` for logging retry attempts.

- **Bug Fixes**
	- Improved delivery status fetching in the `MailgunService` by sanitizing the `messageId`, reducing potential errors in API calls.

- **Documentation**
	- Updated module imports to include the new `RetryNotification` entity, ensuring proper functionality within the notifications system.

### CODE REVIEW

File: apps/api/src/jobs/consumers/notifications/pushSns-notifications.job.consumer.ts [hunk 1]
```diff
@@ -7,13 +7,17 @@
 import { ConfigService } from '@nestjs/config';
 import { PushSnsData, PushSnsService } from 'src/modules/providers/push-sns/push-sns.service';
 import { WebhookService } from 'src/modules/webhook/webhook.service';
+import { RetryNotification } from 'src/modules/notifications/entities/retry-notification.entity';
 import { NotificationQueueProducer } from 'src/jobs/producers/notifications/notifications.job.producer';
 
+
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/api/src/jobs/consumers/notifications/pushSns-notifications.job.consumer.ts]: it already removed in the latest commit

File: apps/api/src/jobs/consumers/notifications/smtp-notifications.job.consumer.ts [hunk 2]
```diff
@@ -8,13 +8,17 @@
 import { Inject, Injectable, forwardRef } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import { WebhookService } from 'src/modules/webhook/webhook.service';
+import { RetryNotification } from 'src/modules/notifications/entities/retry-notification.entity';
 import { NotificationQueueProducer } from 'src/jobs/producers/notifications/notifications.job.producer';
 
+
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/api/src/jobs/consumers/notifications/smtp-notifications.job.consumer.ts]: it already removed in latest commit

File: apps/api/src/modules/notifications/entities/retry-notification.entity.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  @UpdateDateColumn({ name: 'modified_on' })
+  modifiedOn: Date;
+
+  @Column({ name: 'created_by', type: 'varchar', length: 255, nullable: true })
+  createdBy: string;
+
+  @Column({ name: 'modified_by', type: 'varchar', length: 255, nullable: true })
+  modifiedBy: string;
+}
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/api/src/modules/notifications/entities/retry-notification.entity.ts:44]: We can keep the notification_id for more clarity about the notification we talking about 
Also the status here is for the active/inactive record not the retry Status

File: apps/api/src/jobs/consumers/notifications/notification.consumer.ts [hunk 4]
```diff
@@ -71,11 +74,35 @@ export abstract class NotificationConsumer {
...
+      // Save retry attempt record
+      await this.notificationRetryRepository.save({
+        notification,
+        notification_id: notification.id,
+        retryCount: notification.retryCount,
+        retryResult: JSON.stringify({ message: error.message, stack: error.stack }),
+        status: notification.deliveryStatus,
+        createdBy: 'system', // You can replace 'system' with the actual user if available
+        modifiedBy: 'system',
+      });
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/api/src/jobs/consumers/notifications/notification.consumer.ts]: Updated in the latest commit : https://github.com/OsmosysSoftware/osmo-x/pull/308/commits/884b3f510c33af3d28d2be31ed4330c66cf5636c

**Review by xixas — DISMISSED** [review]

**Review by xixas — APPROVED** [review]

**Pull request merged**